scripts/timeline/utils.jl

```diff
@@ -0,0 +1,70 @@
+using Dates
+using Pkg
+
+# Relies on this script being in $STS-PATH/scripts/timeline
+STS_PATH = dirname(dirname(@__DIR__))
+
+# Like @info but also print timestamp & flush immediately
+macro info_extra(msg, rest...)
+    result = quote
+        Base.@info "[$(Dates.now(Dates.UTC))] " * string($(msg)) $(rest...)
+        Base.flush(stdout)
+        Base.flush(stderr)
+    end
+    esc(result)
+end
+
+ensure_dir(dir::AbstractString) = ispath(dir) || mkpath(dir)
+
+function dump(file::AbstractString, what::AbstractString)
```

Comment:
Just in case, Julia has two kinds of syntax for declaring functions, which I find annoying. In my projects I (try to) universally use the other syntax:
```
dump(file::AbstractString, what::AbstractString) = begin
  ...
end
```
(omitting `begin/end` if fits in one expression is acceptable).

But I don't suggest you redo this whole thing in the other style. Just something to keep in mind going forward.